app/src/newdocs/content/tutorials.mdx

```diff
@@ -1,12 +1,239 @@
+
+<div className={docsStyles.centered}>
+  <img src={GetYourAPIKey} />
+</div>
+
+From **Settings > API Keys**, click to add a new key, and give it a name like Tutorial.  Click the meatball menu to copy the new key and click Save Changes to return to your User Page.
+
+<Warning>
+
+**Only use your tutorial API keys for testing and development.** This ensures you don't accidentally modify a live stream or canvas.
+
+</Warning>
+
+<div className={docsStyles.centered}>
+  <img src={SettingsAPICredentials} />
+</div>
 
 </div></ScrollableAnchor>
+<ScrollableAnchor id="publish-to-a-stream"><div>
+
+## Publish to a stream
+
+The easiest way to push data to a stream is to use the [JavaScript client](https://github.com/streamr-dev/streamr-client-javascript), which works in node.js as well as the browser. Client libraries for other languages are on the roadmap. 
+
+You can also try it locally in your browser. Just open this [JS Bin](https://jsbin.com/muqarukubu/edit?js,output), and replace <kbd>MY-STREAM-ID</kbd> and <kbd>MY-API-KEY</kbd> with your Stream ID and API key. Your browser will start producing data to your Stream!
+
+You can also interact with the Streamr API using any HTTP library of your choice. You will find all the details in the <Link to={formatPath(links.docs.api)}>API docs</Link>, but here’s a brief rundown:
+
+You’ll be making <kbd>HTTP POST</kbd> requests to a URL, which contains your Stream ID:
+<CodeSnippet language='json' wrapLines >{`https://www.streamr.com/api/v1/streams/MY-STREAM-ID/data`}</CodeSnippet>
+
+The body of the request will be your data payload in JSON.
+
+Add your API key to the HTTP headers as follows:
+
+<CodeSnippet language='json' wrapLines >{`Authorization: token MY-API-KEY`}</CodeSnippet>
+
+This is how a test request would look in the fabulous [Postman](https://www.getpostman.com/) app, with the URL and Authorization header set:
+
+<div className={docsStyles.centered}>
+  <img src={Postman} />
+</div>
+
+All the different authorization options are explained in the: <Link to={formatPath(links.docs.api)}>API docs</Link>
+
+</div></ScrollableAnchor>
+<ScrollableAnchor id="subscribe-to-data"><div>
+
+## Subscribe to data
+
+Along with Publishing data to a stream, this is best done with the [JavaScript client](https://github.com/streamr-dev/streamr-client-javascript) as mentioned above. 
+
+Here is the given example from the Streamr Javascript client [readme](https://github.com/streamr-dev/streamr-client-javascript):
+
+<CodeSnippet language='javascript' wrapLines showLineNumbers >{ClientSub}</CodeSnippet> 
+
+</div></ScrollableAnchor>
+<ScrollableAnchor id="patterns-for-data-integration"><div>
+
+## Patterns for data integration
+
+There are three distinct patterns for connecting data to Streamr, each with their own pros and cons:
+
+**1.** Pushing from the source (ideal)
+
+**2.** Bridging from a streaming source
+
+**3.** Bridging by polling a source
+
+### 1. Pushing from the source
```

Comment:
This would be a good place to start the actual tutorial and leave out the explainers to others sections.